tofu/geom/_sampling_tools.pyx

```diff
@@ -2216,13 +2210,15 @@ cdef inline void sa_assemble_arrays_unblock(double[:, ::1] part_coords,
     is_vis = <long*> malloc(sz_p * sizeof(long))
     with nogil, parallel(num_threads=num_threads):
         for ii in prange(sz_r):
-            vol_pi = reso_phi[ii] * c_pi
             loc_r = disc_r[ii]
+            vol_pi = loc_r * step_rphi[ii] * c_pi
```

Comment:
this looks like R x Rdphi x Pi

Please confirm:
* Pi is taken from the solid angle formula to save a multiplication inside the innermost loop ?
* step_rphi = R x dphi is the volume element in the phi direction
* Why is there an additional R ? (loc_r) ?

Reply:
yes, `pi` is there only for optimization
`step_rphi` is `dphi` the name comes from the fact that is not the same step for all `r`. Maybe in latex it would be written `$d \varphi_R$`. 
So we still need to multiply by R to obtain `R * dPhi_R`